src/seeds.organization.cpp

```diff
@@ -307,4 +328,136 @@ ACTION organization::subregen(name organization, name account) {
     vote(organization, account, -1 * getregenp(account));
 }
 
+ACTION organization::registerapp(name owner, name organization, name appname, string applongname) {
+    require_auth(owner);
+    check_owner(organization, owner);
+
+    auto orgitr = organizations.find(organization.value);
+    check(orgitr != organizations.end(), "This organization does not exist.");
+
+    auto appitr = apps.find(appname.value);
+    check(appitr == apps.end(), "This application already exists.");
+
+    apps.emplace(_self, [&](auto & app){
+        app.app_name = appname;
+        app.org_name = organization;
+        app.app_long_name = applongname;
+        app.is_banned = false;
+        app.number_of_uses = 0;
+    });
+}
+
+ACTION organization::banapp(name appname) {
+    require_auth(get_self());
+
+    auto appitr = apps.find(appname.value);
+    check(appitr != apps.end(), "This application does not exists.");
+    
+    apps.modify(appitr, _self, [&](auto & app){
+        app.is_banned = true;
+    });
+}
+
+ACTION organization::appuse(name appname, name account) {
+    require_auth(account);
+    check_user(account);
+
+    auto appitr = apps.find(appname.value);
+    check(appitr != apps.end(), "This application does not exists.");
+    check(!(appitr -> is_banned), "Can not use a banned app.");
+    
+    dau_tables daus(get_self(), appname.value);
+    
+    auto dauitr = daus.find(account.value);
+    if (dauitr != daus.end()) {
+        daus.modify(dauitr, _self, [&](auto & dau){
+            dau.number_app_uses += 1;
+        });
+    } else {
+        daus.emplace(_self, [&](auto & dau){
+            dau.account = account;
+            dau.number_app_uses = 1;
+        });
+    }
+
+    apps.modify(appitr, _self, [&](auto & app){
+        app.number_of_uses += 1;
+    });
+}
+
+ACTION organization::cleandaus () {
+    require_auth(get_self());
+
+    // if this function is supposed to be run every 24h,
+    // the get_beginning_of_day_in_seconds will give the current day, but
+    // those values are from the previous day, that's why the timestamp is
+    // from yesterday and not from today
+    uint64_t utc_timestamp_yesterday = get_beginning_of_day_in_seconds() - utils::seconds_per_day;
+
+    auto appitr = apps.begin();
+    while (appitr != apps.end()) {
+        if (appitr -> is_banned) {
+            appitr++;
+            continue; 
+        }
+        action clean_dau_action(
+            permission_level{get_self(), "active"_n},
+            get_self(),
+            "cleandau"_n,
+            std::make_tuple(appitr -> app_name, utc_timestamp_yesterday, (uint64_t)0)
+        );
+
+        transaction tx;
+        tx.actions.emplace_back(clean_dau_action);
+        tx.delay_sec = 1;
+        tx.send((appitr -> app_name).value + 1, _self);
+
+        appitr++;
+    }
+}
+
+ACTION organization::cleandau (name appname, uint64_t timestamp, uint64_t start) {
```

Comment:
Is cleandaus supposed to be called by scheduler?

Reply:
Yes `cleandaus` is the function that should be called daily, and then `cleandaus` will call `cleandau` for each app.

I committed a change that stores the current day and then the function `cleandaus` can be called at any time and it will just store the information of the previous days, just in case the `cleandaus` funtion fails for some reason it could be called at any time. Also when using `appuse` action, it also check the day, in case of any error from `cleandaus`, `appuse` action will store the old info for that user so the information is consistent. 

I added the `cleandaus` action to the scheduler, I configured it to be called every 12 hours just in case.

Reply:
Cool!